src/Search/SearchWithSuggestions/SearchWithSuggestions.js

```diff
@@ -171,37 +260,187 @@ class SearchWithSuggestions extends PureComponent {
+              suggestionContent={suggestionContent}
+              isSearching={isSearching}
+              maxSuggestions={maxSuggestions}
+            />
           </Panel>
         )}
       </div>
     )
   }
 }
 
+const SuggestionItems = ({
+  suggestions,
+  cursor,
+  onSuggestionSelect,
+  suggestionContent,
+  isSearching,
+  maxSuggestions,
+  onViewAllResults,
+  searchTerm
+}) => {
+  const getSliced = data =>
+    maxSuggestions ? data.slice(0, maxSuggestions) : data
+
+  if (isGroups(suggestions)) {
+    const noData = getLengthOfSuggestions(suggestions, maxSuggestions) === 0
+    const types = Object.keys(suggestions)
+
+    let fromCounter = 0
+
+    return (
+      <>
+        {!noData && (
+          <ViewAllResults
+            searchTerm={searchTerm}
+            onViewAllResults={onViewAllResults}
+            suggestions={suggestions}
+          />
+        )}
+        {!noData &&
+          types.map((key, index) => {
+            const { label, options } = suggestions[key]
+
+            const formattedOptions = getSliced(options)
+
+            if (!formattedOptions.length) {
+              return null
+            }
+
+            fromCounter += formattedOptions.length
+
+            return (
+              <Fragment key={key}>
+                {label && <div className={styles.groupLabel}>{label}</div>}
+                <SuggestionItemsList
+                  fromCounter={fromCounter - formattedOptions.length}
+                  suggestions={formattedOptions}
+                  cursor={cursor}
+                  onSuggestionSelect={selected =>
+                    onSuggestionSelect([key, selected])
+                  }
+                  suggestionContent={suggestionContent}
+                />
+                {index !== types.length - 1 && (
+                  <div className={styles.divider} />
+                )}
+              </Fragment>
+            )
+          })}
+        {noData && <NoResults isSearching={isSearching} />}
+      </>
+    )
+  } else {
+    const sliced = getSliced(suggestions)
+    return sliced.length > 0 ? (
+      <>
+        <ViewAllResults
+          searchTerm={searchTerm}
+          onViewAllResults={onViewAllResults}
+          suggestions={suggestions}
+        />
+        <SuggestionItemsList
+          suggestions={sliced}
+          cursor={cursor}
+          onSuggestionSelect={onSuggestionSelect}
+          suggestionContent={suggestionContent}
+        />
+      </>
+    ) : (
+      <NoResults isSearching={isSearching} />
+    )
+  }
+}
+
+const ViewAllResults = ({ searchTerm, onViewAllResults, suggestions }) => {
+  if (!searchTerm || !onViewAllResults) {
+    return null
+  }
+
+  return (
+    <>
+      <div
+        className={styles.viewAllResults}
+        onMouseDown={() => {
+          onViewAllResults(searchTerm, suggestions)
+        }}
+      >
+        View all results for “{searchTerm}”
+      </div>
+      <div className={styles.divider} />
```

Comment:
hm, do we really need divider as div element? Is it possible to do it as additional class?

Reply:
Changed to class